### PR TITLE
Add new columns for tracking bounces

### PR DESCRIPTION
**Pull request by andrewleith**
# Summary | Résumé

This PR adds new columns to the `notifications` and `notification_history` tables:
- feedback_type
- feedback_subtype
- ses_feedback_id
- ses_feedback_date

It also adds the properties to the models as well as some constants for dealing with the values.

### CODE REVIEW

File: migrations/versions/0426_add_bounce_type_columns.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+import sqlalchemy as sa
+from alembic import op
+
+
+# option 1
+def upgrade():
+    # 1 - add feedback types to notifications/notifications_history table
+    feedback_types = sa.Enum("hard-bounce", "soft-bounce", name="notification_feedback_types")
+    feedback_types.create(op.get_bind())
+    op.add_column("notifications", sa.Column("feedback_type", feedback_types, nullable=True))
```
Comment (jzbahrai) [inline migrations/versions/0426_add_bounce_type_columns.py:22]: can we have an index on this? and can you add it concurrently?

**Review by jzbahrai — CHANGES_REQUESTED** [review]

File: migrations/versions/0426_add_bounce_type_columns.py (hunk 1 above)
Comment (andrewleith) [reply migrations/versions/0426_add_bounce_type_columns.py:22]: Yes! I just added a second migration for this: https://github.com/cds-snc/notification-api/pull/1768/files#diff-772bb11b0fb474d256bb93ee066ab549088ef7989abef084cff8edfedd2e3633

File: migrations/versions/0427_add_bounce_type_indices.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+Revises: 0426_add_bounce_type_columns
+Create Date: 2017-04-25 11:34:43.229494
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = "0427_add_bounce_type_indices"
+down_revision = "0426_add_bounce_type_columns"
+
+import sqlalchemy as sa
```
[the remark this answers is not in the document]
Comment (jzbahrai) [reply migrations/versions/0427_add_bounce_type_indices.py:13]: @andrewleith can you add this migration to the previous one? we should be good to run it on the same migration as it is a new column

[the remark this answers is not in the document]
Comment (andrewleith) [reply migrations/versions/0427_add_bounce_type_indices.py:13]: It wouldn't let me.  Complained about it being part of a transaction?

[the remark this answers is not in the document]
Comment (andrewleith) [reply migrations/versions/0427_add_bounce_type_indices.py:13]: It wouldn't let me.  Complained about it being part of a transaction:
```
sqlalchemy.exc.InternalError: (psycopg2.errors.ActiveSqlTransaction) CREATE INDEX CONCURRENTLY cannot run inside a transaction block
```

**Review by jzbahrai — APPROVED** [review]

**Pull request merged**